### PR TITLE
wxWidgets intial recipe

### PR DESCRIPTION
**Pull request by gmeeker**
Specify library name and version:  **wxwidgets/3.2.3**

Implements #5190 #19991 #20092
Requested by #621

This is a new recipe to CCI. It's an updated version from bincrafters but ported to Conan v2 with additional fixes.

https://github.com/bincrafters/community/tree/main/recipes/wxwidgets

Also unmerged GTK3 work from here (using GTK from Conan recipe):

https://github.com/bincrafters/community/pull/1434

GTK support is currently a draft but macOS and Windows seem usable. Any feedback appreciated. For example, do we need anything older than 3.2.3 here?

---

- [X] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [X] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [X] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by gmeeker** [comment]
I can't see why the last build failed. conan-center-bot didn't update. The previous commit was successful and the latest builds for me on macOS, Windows, and Linux.
I don't have time to pursue this, but it currently only builds GTK with gtk/system (which defaults to GTK 2 unfortunately). The user can override to GTK 3 but CCI requires prebuilt dependencies. This is unfortunate for WebKitGTK (which will be a system dependency without a lot of work), but modern distros only provide the GTK 3 version.
Otherwise I think this is fairly complete. I suppose it might be nice to use Conan components.

**Comment by valgur** [comment]
The non-system versions of GTK can be used as soon as #21123 gets merged. It's currently blocked by a similar migration PR for `pango`.

**Comment by vpicaver** [comment]
What needs to be done to get this merged?

File: recipes/wxwidgets/all/conandata.yml
Comment (valgur) [inline recipes/wxwidgets/all/conandata.yml:13]: Don't include additional versions unless there's a specific need for them.

File: recipes/wxwidgets/all/conanfile.py
Comment (valgur) [inline recipes/wxwidgets/all/conanfile.py:131]: Needs to be replaced with the `cairo` package from Conan, probably.

File: recipes/wxwidgets/all/conanfile.py
Comment (valgur) [inline recipes/wxwidgets/all/conanfile.py:36]: Should get rid of the `gtk/system` dependency now.

File: recipes/wxwidgets/all/conanfile.py
Comment (valgur) [inline recipes/wxwidgets/all/conanfile.py:167]:
```suggestion
            self.requires("libcurl/[>=7.78.0 <9]")
```

File: recipes/wxwidgets/all/conanfile.py
Comment (valgur) [inline recipes/wxwidgets/all/conanfile.py:169]:
```suggestion
            self.requires("libpng/[>=1.6 <2]")
```

File: recipes/wxwidgets/all/conanfile.py
Comment (valgur) [inline recipes/wxwidgets/all/conanfile.py:195]: Set these using `self.option[...].xyz` or `self.requires(..., options={...})` as well.

File: recipes/wxwidgets/all/conanfile.py
Comment (valgur) [inline recipes/wxwidgets/all/conanfile.py:286]:
```suggestion
        tc.variables["wxUSE_FS_INET"] = self.options.fs_inet
        tc.cache_variables["CMAKE_POLICY_DEFAULT_CMP0077"] = "NEW"
```
For `tc.variables` to work correctly.

**Review by valgur — CHANGES_REQUESTED** [review]
Thanks! That's definitely a very challenging library to package.

Although unlikely to work currently due to version conflicts, the recipe should switch to using `cairo` and `gtk` from Conan.

**Comment by Codinablack** [comment]
This looks amazing! I just want to subscribe to this so I know when it is done, and so I say thank you for your time in making it happen, I know there are many wanting to see this!

File: recipes/wxwidgets/all/conanfile.py
Comment (gmeeker) [reply recipes/wxwidgets/all/conanfile.py:196]: Added `self.requires(..., options={...})`

**Comment by gmeeker** [comment]
1. Using GTK from Conan is still not working because that recipe doesn't support Conan v2 yet.
2. I'm not sure what would happen if you mix a Conan built GTK with system WebKitGTK (and writing a recipe for that looks even more daunting).
3. wxWidgets is still GTK3 only but I don't think that affects any of this.
4. I'm not able to use Conan cairo with system GTK. I'm getting missing glib symbols (maybe mixing system + built glibs).

So I'm not sure if this will be solved anytime soon.

**Comment by czoido** [comment]
Hi @gmeeker,
Thanks a lot for the contribution, I'm pushing some changes to simplify the options (if it's needed in the future we can always put them back), relying on defaults for the moment and also removing the gtk option until we have a gtk recipe compatible with Conan 2 (I'm working on some changes for the [PR](https://github.com/conan-io/conan-center-index/pull/21123) open, so I hope it will be soon)

**Review by memsharded — APPROVED** [review]

**Review by czoido — APPROVED** [review]

**Comment by SpaceIm** [comment]
Why is there ninja in build requirements? Putting ninja in build requirement of a recipe doesn't mean that it will be used as CMake generator. Build pass, and ninja is not used at all as we can see in build logs, so I think it can be removed.

File: recipes/wxwidgets/all/conanfile.py
Comment (SpaceIm) [inline recipes/wxwidgets/all/conanfile.py:101]: There is libsecret in conancenter.

I'm surprised by this recipe. I thought it was forbidden to have system requirements explicitly requested in conancenter (except for a "system" recipe).

Comment (Croydon) [reply recipes/wxwidgets/all/conanfile.py:101]: Same question. How did this get past hook checks and human reviews?

**Pull request merged**
